frontend/web/components/pages/EnvironmentSettingsPage.tsx

```diff
@@ -0,0 +1,891 @@
+    openModal(
+      title,
+      <ConfirmToggleEnvFeature
+        description={'Are you sure that you want to change this value?'}
+        feature={Format.enumeration.get(environmentProperty)}
+        featureValue={environmentPropertyValue}
+        onToggleChange={() => {
+          updateCurrentEnv({ [environmentProperty]: environmentPropertyValue }, true)
+          closeModal()
+        }}
+      />,
+      'p-0 modal-sm',
+    )
+  }
+
+  const onEnableVersioning = () => {
+    if (!currentEnv?.api_key) return
+    openConfirm({
+      body: 'This will allow you to attach versions to updating feature values and segment overrides. Note: this may take several minutes to process',
+      onYes: () => {
+        enableFeatureVersioning(store, {
+          environmentId: currentEnv?.api_key,
+        }).then(() => {
+          toast(
+            'Feature Versioning Enabled, this may take several minutes to process.',
+          )
+          updateCurrentEnv({
+            enabledFeatureVersioning: true,
+          }, false, true)
+        })
+      },
+      title: 'Enable "Feature Versioning"',
+    })
+  }
+
+  return (
+    <div className='app-container container'>
+      <ProjectProvider
+        onRemoveEnvironment={onRemoveEnvironment}
+        id={match.params.projectId}
+        onRemove={onRemove}
+        onSave={onSave}
+      >
+        {({ deleteEnv, isLoading, isSaving, project }) => {
+          const env = _.find(project?.environments, {
+            api_key: match.params.environmentId,
+          })
+          if (
+            (env &&
+              typeof env?.minimum_change_request_approvals ===
+              'undefined') ||
+            env?.api_key !== match.params.environmentId
+          ) {
+            setTimeout(() => {
+              const minimumChangeRequestApprovals = Utils.changeRequestsEnabled(env?.minimum_change_request_approvals)
+              updateCurrentEnv({
+                allow_client_traits: !!env?.allow_client_traits,
+                banner_colour: env?.banner_colour || Constants.tagColors[0],
+                banner_text: env?.banner_text,
+                hide_disabled_flags: env?.hide_disabled_flags || false,
+                hide_sensitive_data: !!env?.hide_sensitive_data,
+                minimum_change_request_approvals: minimumChangeRequestApprovals
+                  ? env?.minimum_change_request_approvals
+                  : null,
+                name: env?.name || "",
+                use_identity_composite_key_for_hashing:
+                  !!env?.use_identity_composite_key_for_hashing,
+                use_identity_overrides_in_local_eval:
+                  !!env?.use_identity_overrides_in_local_eval,
+                use_v2_feature_versioning: !!env?.use_v2_feature_versioning,
+              }, false, true)
+            }, 10)
+          }
+
+          return (
+            <>
+              <DirtyFormModal />
+              <PageTitle title='Settings' />
```

Comment:
Overall, I believe the page could be broken up in a lot of pieces and sub-form that would be in charge of partially update necessary part of the environment:

- Each tab could be its own component.
- General settings could be an isolated form (especially as it is the only one being submitted)
- Same for the switch part etc

Reply:
Yep agree that separating the form and other components makes sense.